ginga/misc/plugins/SaveImage.py

```diff
@@ -0,0 +1,477 @@
+        for key in func():
+            iminfo = self.chinfo.get_image_info(key)
+            path = iminfo.get('path')
+            idx = iminfo.get('idx')
+            t = iminfo.get('time_modified')
+
+            if path is None:  # Special handling for generated buffer, eg mosaic
+                infile = key
+                is_fits = True
+            else:
+                infile = os.path.basename(path)
+                infile_ext = os.path.splitext(path)[1]
+                infile_ext = infile_ext.lower()
+                is_fits = False
+
+                if 'fit' in infile_ext:
+                    is_fits = True
+
+            # Only list FITS files unless it is Ginga generated buffer
+            if not is_fits:
+                continue
+
+            # Only list modified buffers
+            if mod_only and t is None:
+                continue
+
+            # More than one ext modified, append to existing entry
+            if infile in treedict:
+                if t is not None:
+                    treedict[infile].extlist.add(idx)
+                    elist = sorted(treedict[infile].extlist)
+                    treedict[infile].MODEXT = ';'.join(
+                        map(self._format_extname, elist))
+
+            # Add new entry
+            else:
+                if t is None:
+                    s = ''
+                    extlist = set()
+                else:
+                    s = self._format_extname(idx)
+                    extlist = set([idx])
+                treedict[infile] = Bunch.Bunch(
+                    IMAGE=infile, MODEXT=s, extlist=extlist, path=path)
+
+        self.treeview.set_tree(treedict)
+
+        # Resize column widths
+        n_rows = len(treedict)
+        if n_rows == 0:
+            self.w.status.set_text('Nothing available for saving')
+        elif n_rows < self.settings.get('max_rows_for_col_resize', 5000):
+            self.treeview.set_optimal_column_widths()
+            self.logger.debug('Resized columns for {0} row(s)'.format(n_rows))
+
+        return
+
+    def _check_chinfo(self):
+        """Special handling for when plugin is started from global plugin
+        menu.
+
+        This locks the plugin to the active channel when it starts.
+        To change the associated channel in global plugin space, user
+        needs to manual close and reload the plugin. Doing this
+        dynamically inside redo() messes up the behavior in
+        local plugin space.
+
+        """
+        self.chinfo = self.fv.get_channelInfo()
+
+        if self.chinfo is None:
+            raise ValueError('No channel available')
+
+        self.chname = self.chinfo.name
+
+    def keys_from_datasrc(self):
+        """Yield back keys for image listing from Ginga's data cache."""
+        for key in self.chinfo.datasrc.sortedkeys:
+            yield key
+
+    def keys_from_contents(self):
+        """Yield back keys for image listing  from ``Contents`` global plugin.
+        """
```

Comment:
I think you should be able to get all images by scanning the channel lists from `Control.py`.  That way if there is a configuration where the user doesn't use `Contents`, or has replaced it with their own, it would still work.  Code would be something like:
```python
chnames = self.fv.get_channelNames()
imnames = []
for chname in chnames:
    channel = self.fv.get_channelInfo(chname)
    imnames.extend(channel.get_image_names())
return imnames
```
You could use `yield` here if you want to write it that way, but since the list is being created, consumed and then dropped I'm not sure it really makes any difference except that the non-`yield` version should be faster.